gui/src/app/state/transactions.rs

```diff
@@ -146,23 +147,23 @@ impl State for TransactionsPanel {
                 if let Some(tx) = &self.selected_tx {
                     if tx.fee_amount.is_some() {
                         let tx = tx.clone();
-                        let txid = tx.tx.txid();
+                        let outpoints: Vec<_> = (0..tx.tx.output.len())
+                            .map(|vout| {
+                                OutPoint::new(
+                                    tx.tx.txid(),
+                                    vout.try_into()
+                                        .expect("number of transaction outputs must fit in u32"),
+                                )
+                            })
+                            .collect();
                         return Command::perform(
                             async move {
                                 daemon
-                                    // TODO: filter for spending coins when this is possible:
-                                    // https://github.com/wizardsardine/liana/issues/677
-                                    .list_coins()
+                                    .list_coins(&[CoinStatus::Spending], &outpoints)
                                     .map(|res| {
                                         res.coins
                                             .iter()
-                                            .filter_map(|c| {
-                                                if c.outpoint.txid == txid {
-                                                    c.spend_info.map(|info| info.txid)
-                                                } else {
-                                                    None
-                                                }
-                                            })
+                                            .map(|c| c.spend_info.expect("coin is spending").txid)
```

Comment:
I would prefer we keep a filter_map and no expect

Reply:
I've changed this to filter_map.